src/blockserver/backend/transfer.py

```diff
@@ -157,45 +158,47 @@ def delete(self, storage_object):
         return size
 
 
-files = {}
+class LocalTransfer(AbstractTransfer):
 
-
-class DummyTransfer(AbstractTransfer):
-
-    def __init__(self, cache):
+    def __init__(self, basedir, cache):
         super().__init__(cache)
-        self._tempdir = tempfile.mkdtemp()
+        self.basepath = Path(basedir)
 
     def get_size(self, storage_object: StorageObject) -> int:
-        try:
-            cached = self._from_cache(storage_object)
-        except KeyError:
-            try:
-                return os.path.getsize(
-                    files[file_key(storage_object)].local_file)
-            except KeyError:
-                return None
-        else:
-            return cached.size
+        return getattr(self.meta(storage_object), 'size', 0)
 
     def store(self, storage_object: StorageObject) -> Tuple[StorageObject, int]:
         old_size = self.get_size(storage_object)
         if old_size is None:
             old_size = 0
         new_size = os.path.getsize(storage_object.local_file)
-        new_path = os.path.join(self._tempdir, file_key(storage_object))
-        dirname = os.path.dirname(new_path)
-        os.makedirs(dirname, exist_ok=True)
-        if new_size != 0:
-            shutil.copyfile(storage_object.local_file, new_path)
-        else:
-            Path(new_path).touch()
+        target_path = self.basepath / file_key(storage_object)
+        target_path.parent.mkdir(parents=True, exist_ok=True)
+
+        # Three ways we create the final file:
+        try:
+            # (1) just rename(2) it (fast: no data copy, but only inside the same FS)
```

Comment:
oh and that `if not os_error.errorno in...` would be faster to read if it would get a plausible name like
`if not is_real_file(os_error)` or some name that explains what's checked here